### ISSUE TITLE
Document why scroll views should not have accessibility labels

### ISSUE THREAD

**Issue by phatmann** [kif-framework/KIF#592]
"Using accessibility labels for scroll views like UITableView is not recommended, since they hide access to the accessibility labels of their subviews."

See discussions in #580 and #569.

**Comment by SebastianMecklenburg** [comment]
I just tried to access a table view cell by it's accessibility label and it worked fine, although it's table view also had an accessibility label. I wrote me a function to output the view hierarchy along with the accessibility labels (I posted it to the google group some time ago). Below is the output. I can get the UITableview and the UILabel by their accessibility labels just fine. Maybe the problem exists only under certain circumstances?

```
|	|	UITableView, label: MetaDataTableView
|	|	|	UITableViewWrapperView
|	|	|	|	CWMetaDataTableViewCell
|	|	|	|	|	UITableViewCellContentView
|	|	|	|	|	|	UILabel, label: Beschreibung
|	|	|	|	|	|	UILabel, label: 
|	|	|	|	|	_UITableViewCellSeparatorView
```
```
    //both of these return the correct view
    UIView* labelView = [tester waitForViewWithAccessibilityLabel:@"Beschreibung"];
    UITableView* tv = [tester waitForViewWithAccessibilityLabel:@"MetaDataTableView"];
````

**Comment by phatmann** [comment]
My personal experience using UIAutomation was that a child view is not accessible if the parent scroll view has an accessibility label. However, I could find nothing online to back up this claim.

The actual reason that KIF uses an identifier instead of a label for scroll views was provided by @bnickel in [this comment](https://github.com/kif-framework/KIF/issues/243#issuecomment-50545987):

> The reason accessibilityIdentifier is favored for scrolling is because UIScrollView and UITableView 
> don't actually announce or otherwise use their accessibilityLabel in terms of actually accessibility 
> functionality. In essence the deprecated methods were using accessibilityLabel as an 
> accessibilityIdentifier, just a searchable string for finding the control.

So this is what I need to document. Thanks for correcting the record, @SebastianMecklenburg!

**Comment by SebastianMecklenburg** [comment]
Ahh, OK. Activating VoiceOver really gets quite confusing if the scroll views have accessibility labels. An alternative, as proposed [here](http://www.raywenderlich.com/61419/ios-ui-testing-with-kif) would be to use labels and to use a compiler switch to set them only for the test build. 
That's what I find a bit difficult about using accessibility labels for testing, it mixes two different technologies and we have to make sure the labels work equally well for testing and for VoiceOver. Not that I have a better idea though...

**Comment by SebastianMecklenburg** [comment]
Maybe I have a better idea after all. What I dislike most about the accessibility concept is that the user of the KIF library has to know when an identifier is being used and when a label. I would consider this an implementation detail that should be encapsulated. So I wrote me a method in a KIFUITestActor catgory that looks like
```
    UIView *v = [tester waitForView:@"Login"];
```

and the (not yet very much tested) implementation is

```
- (UIView *)waitForView:(NSString*)labelOrIdentifier {
    
    __block UIView* view = nil;
    [self runBlock:^KIFTestStepResult(NSError **error) {
        view = [self getView:labelOrIdentifier];
        return view != nil ? KIFTestStepResultSuccess : KIFTestStepResultWait;
    }];
    return view;
}

-(UIView*)getView:(NSString*)labelOrIdentifier {
    NSError *error = nil;
    UIView *view;
    BOOL success = [UIAccessibilityElement accessibilityElement:nil view:&view withLabel:labelOrIdentifier value:nil traits:0 tappable:NO error:&error];
    if(!success) {
        NSPredicate* predicate = [NSPredicate predicateWithFormat:@"accessibilityIdentifier = %@", labelOrIdentifier];
        success = [UIAccessibilityElement accessibilityElement:nil view:&view withElementMatchingPredicate:predicate tappable:NO error:&error];
    }
    if(success && view) {
        return view;
    }
    return nil;
}
```

Now I can change my accessibility labels to identifiers as I wish without having to update my test code. Also it looks nicer, IMO :-) 
I find KIF supercool and it works great, I just noticed that I spend most of my time trying to access the right UI elements, for example "the button on the right in the second segment header in the currently visible collection view". I'm not very experienced using KIF, but I guess that is a not so obvious task even for KIF veterans. Having to distinguish between accessibilityLabels and -Identifiers complicates things even further. If we could find some more ways to access UI elements it would make KIF quite a bit easier to use, I think. Maybe it would also help (at least partly) with the internationalisation problem that recently came up in the google group.
